### PR TITLE
Refining Cpi details

### PR DESCRIPTION
**Pull request by gberche-orange**
Refining stemcell descriptions.
- Linked to stemcell page
- Refined description of light stemcells reusing material from 
https://groups.google.com/a/cloudfoundry.org/d/msg/bosh-users/3RZsYU03U5g/4g4ikKoWkxkJ
- Mention future universal stemcells

Added pointers to overview of component interactions and calls sequenting

### CODE REVIEW

File: cpi-api-v1.html.md.erb [hunk 1]
```diff
@@ -14,12 +14,19 @@ resource_pools:
   cloud_properties: {instance_type: r3.8xlarge}
 ```
 
+For an overview of the sequence of CPI calls, the following resources are helpful:
+- [bosh components](bosh-components.html) and its example component interaction during new vm creation
+- typical vm instanciation by bosh-init described in the [bosh-init architecture doc](https://github.com/cloudfoundry/bosh-init/blob/master/docs/architecture.md#deploy-command-flow) where calls to the cpi are mentionned as `cloud.`
```
Comment (voelzmo) [inline cpi-api-v1.html.md.erb]: typos: instantiation, mentioned

File: cpi-api-v1.html.md.erb [hunk 2]
```diff
@@ -14,12 +14,19 @@ resource_pools:
...
+- typical vm instanciation by bosh-init described in the [bosh-init architecture doc](https://github.com/cloudfoundry/bosh-init/blob/master/docs/architecture.md#deploy-command-flow) where calls to the cpi are mentionned as `cloud.` 
+
 ---
 ## <a id="stemcells"></a> Stemcell management
 
 ### <a id="create-stemcell"></a> `create_stemcell`
 
-Creates a reusable VM image in the IaaS from the stemcell image. It's used later for creating VMs. For example AWS CPI creates an AMI and returns AMI ID.
+Creates a reusable VM image in the IaaS from the [stemcell](stemcell.html) image. It's used later for creating VMs. For example AWS CPI creates an AMI and returns AMI ID. 
+Stemcell tarballs are currently specific to each Iaas/CPI (e.g. include cloud-init or hypervisor specific drivers). In the future, stemcells will become [universal](https://www.pivotaltracker.com/epic/show/927178) and CPI will bring in the Iaas specific bits before creating the VM image.
```
Comment (voelzmo) [inline cpi-api-v1.html.md.erb]: It's great that there exists an epic for that - however, it doesn't seem to contain any stories at all. Also, https://github.com/cloudfoundry/bosh-notes doesn't contain anything about universal stemcells. If it is that unspecific at the current point in time - should we leave that out of here?

File: cpi-api-v1.html.md.erb [hunk 3]
```diff
@@ -14,12 +14,19 @@ resource_pools:
...
 ---
 ## <a id="stemcells"></a> Stemcell management
 
 ### <a id="create-stemcell"></a> `create_stemcell`
 
-Creates a reusable VM image in the IaaS from the stemcell image. It's used later for creating VMs. For example AWS CPI creates an AMI and returns AMI ID.
+Creates a reusable VM image in the IaaS from the [stemcell](stemcell.html) image. It's used later for creating VMs. For example AWS CPI creates an AMI and returns AMI ID. 
+Stemcell tarballs are currently specific to each Iaas/CPI (e.g. include cloud-init or hypervisor specific drivers). In the future, stemcells will become [universal](https://www.pivotaltracker.com/epic/show/927178) and CPI will bring in the Iaas specific bits before creating the VM image. 
+ 
+In the meantime, if you're developping a CPI for a new IaaS, you may consider reusing one of the officially generated stemcells, or forking the stemcell builder in bosh repo.
```
Comment (voelzmo) [inline cpi-api-v1.html.md.erb]: typo: developing

**Comment by gberche-orange** [comment]
thanks @voelzmo for catching typos, sorry about that. Concerning universal stemcells, I mentionned it because I was originally confused after having watched the [bosh bootstraping](https://www.youtube.com/watch?v=W-_l8051YO8&list=PLhuMOCWn4P9g-UMN5nzDiw78zgf5rJ4gR&index=38) session into thinking external cpis supported universal stemcells. Adding pointer to placeholder epic/story avoids this confusion while allowing to track progress on this. 

I'm fine to remove this if there is a concensus that this brings more confusion than good.

**Comment by cppforlife** [comment]
I've moved some things around and merged.

**Comment by gberche-orange** [comment]
thanks @cppforlife for merging and for the additional details related to building a stemcell for a new cloud.

I wonder whether you saw my pending questions onto https://lists.cloudfoundry.org/archives/list/cf-bosh@lists.cloudfoundry.org/message/ZDS5YZIYSKVOYKBI4WMGQYSEXTOGVXJA/ ? 

As we proceed our [java-based cloudstack cpi implementation](https://github.com/cloudfoundry-community/bosh-cloudstack-cpi-core/commits/master), and proceeding reverse engineering for some parts we did not understand from docs, would it make sense for me to to contribute an initial version of the "Interactions between CPIs and BOSH Agent" listed in the TOC or is it something you're planning soon ?

**Pull request closed without merging**